src/pages/api/events/all/index.ts

```diff
@@ -97,17 +99,24 @@ export default async function handler(
       const results: ModifiedResult[] = [];
       if (keywordSearch.count > 0) {
         for (const edge of keywordSearch.edges) {
-          results.push({
-            id: edge.node.result.id,
-            title: edge.node.result.title,
-            eventUrl: edge.node.result.eventUrl,
-            description: edge.node.result.description,
-            venue: edge.node.result.venue,
-            imageUrl: edge.node.result.image.baseUrl,
-            dateTime: convertDate(edge.node.result.dateTime),
-          });
+          if (
+            edge.node.result.venue &&
+            edge.node.result.venue?.address !== ""
```

Comment:
Instead of removing Online events, we can put multiple pins around the location.
ref: https://stackoverflow.com/questions/20490654/more-than-one-marker-on-same-place-markerclusterer

Reply:
Online events usually don't have an address, and longitude and latitude are irrelevant. I'm not quite sure about the suggestion to place multiple pins around the location. Could you clarify what you mean by that?

Reply:
@SachiGoto I found online events have the same longitude and latitude so I though it's better to keep these data and show events on map instead of removing online events from events. 
The purpose of the suggestion is to show all pins for online events otherwise we only see one pin hiding all other online events.
But I think it's better to override longitude and latitude for online (like Vancouver: Latitude: 49.246292, Longitude: -123.116226) because the original values are too far from Vancouver and users can't find them.
For example: 
```typescript
if (keywordSearch.count > 0) {
        for (const edge of keywordSearch.edges) {
          if (edge.node.result.venue) {
            results.push({
              id: edge.node.result.id,
              title: edge.node.result.title,
              eventUrl: edge.node.result.eventUrl,
              description: edge.node.result.description,
              venue: {
                id: edge.node.result.venue.id,
                name: edge.node.result.venue.name,
                address: edge.node.result.venue.address || "Online",
                lat: edge.node.result.venue.lat || 49.2827,
                lng: edge.node.result.venue.lng || -123.1207,
              },
              imageUrl: edge.node.result.image.baseUrl,
              dateTime: convertDate(edge.node.result.dateTime),
            });
          }
        }
      }
// Or
if (keywordSearch.count > 0) {
        for (const edge of keywordSearch.edges) {
          results.push({
            id: edge.node.result.id,
            title: edge.node.result.title,
            eventUrl: edge.node.result.eventUrl,
            description: edge.node.result.description,
            venue: {
              id: edge.node.result.venue?.id || "12345678",
              name: edge.node.result.venue?.name || "temp name",
              address: edge.node.result.venue?.address || "Online",
              lat: edge.node.result.venue?.lat || 49.2827,
              lng: edge.node.result.venue?.lng || -123.1207,
            },
            imageUrl: edge.node.result.image.baseUrl,
            dateTime: convertDate(edge.node.result.dateTime),
          });
        }
      }
```
The suggestion came from the concern that we can't recognize map pins for online events if they have the same location data. With the code in the link we can get location data slightly different from each other. But this is just one suggestion and we might have a better solution for that.

Btw I think we can discuss how to real with data without `venue`. I find one of them are private event but not sure if that relates to have empty `venue`

Reply:
mm I see. I do have some concerns about displaying online events on the map, though. My concern is that it might lead to confusion, as people could mistake the pinned locations for in-person events. But, I do agree with the idea of retaining online events. Maybe we could consider incorporating a sidebar that lists all the online events separately?

Reply:
Oh I didn't think of that! I think it's a good idea to have a list, I also thought that pointing somewhere in the ocean would work?

If we really want to filter out online events, I would use add another filter `eventType` to fetch physical events only. But I think we can keep the code for now.
https://www.meetup.com/api/schema/#SearchConnectionFilter
https://www.meetup.com/api/schema/#EventType

Reply:
Oh, nice! To me, it makes sense to separate physical and online events and show pins for the in-person events on the map. We could consider listing the online events on the sidebar or exploring other options (there might be a better approach). Personally, I find it a bit unclear to display temporary pins on the map for online events...
but we can talk about it  during our next meeting..?